docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -277,15 +277,40 @@ in the partition::
     DELETE FROM iceberg.testdb.customer_orders
     WHERE country = 'US' AND customer = 'Freds Foods'
 
-Rolling back to a previous snapshot
------------------------------------
+Snapshots
+---------
 
 Iceberg supports a "snapshot" model of data, where table snapshots are
 identified by an snapshot IDs.
 
-The connector provides a system snapshots table for each Iceberg table.  Snapshots are
-identified by BIGINT snapshot IDs.  You can find the latest snapshot ID for table
-``customer_orders`` by running the following command::
+The connector provides a system table containing snapshot information for every
+Iceberg table. Snapshots are identified by ``BIGINT`` snapshot IDs.
+You can find the snapshot IDs for table ``customer_orders`` by running the
+following command::
+
+    SELECT snapshot_id FROM iceberg.testdb."customer_orders$snapshots" ORDER BY committed_at DESC
+
+Snapshot queries
+^^^^^^^^^^^^^^^^
+
+The snapshots of an Iceberg table can be queried by using the following command::
+
+   SELECT * FROM iceberg.testdb."customer_orders@8954597067493422955"
+
+In the command above, the special notation ``"customer_orders@8954597067493422955"``
+is used to point to a specific snapshot of the ``customer_orders`` Iceberg table.
+The snapshot queries can be also used with special tables::
```

Comment:
Nit: maybe `Querying by snapshot ID can also be used with the other [special] metadata tables`.